app/models/proxy.rb

```diff
@@ -559,12 +560,16 @@ def proxy
   class PolicyConfig
     include ActiveModel::Validations
 
+    attr_reader :init_attributes
     attr_accessor :name, :version, :configuration, :enabled
-    validates :name, :version, presence: true
-    validate :configuration_is_object
+
+    validates :name, :version, presence: true, if: :attributes_are_hash?
+    validate :configuration_is_object, if: :attributes_are_hash?
+    validate :attributes_are_hash
 
     def initialize(attributes = {})
-      self.attributes = attributes
+      @init_attributes = attributes
```

Comment:
`#initialize` should check the argument passed instead of validating them, wdyt?

Reply:
okok, that could be also the way.